ui/app/src/modules/Content/Publish/PublishDialog.tsx

```diff
@@ -585,7 +625,17 @@ function PublishDialogWrapper(props: PublishDialogProps) {
       onClickShowAllDeps={showAllDependencies}
       apiState={apiState}
       classes={useStyles()}
-      showEmailCheckbox={!myPermissions.includes('publish')}
+      showEmailCheckbox={!myPermissions.includes('publish') || dialog.requestApproval}
+      showRequestApproval={myPermissions.includes('publish')}
+      submitLabel={
+        dialog.scheduling === 'custom' ? (
+          <FormattedMessage id="requestPublishDialog.schedule" defaultMessage="Schedule" />
+        ) : !myPermissions.includes('publish') || dialog.requestApproval ? (
```

Comment:
There's quite a few instances of `myPermissions.includes('publish')` being executed all over the place. Please store in a var and use instead of constantly iterating through the array.